homeassistant/loader.py

```diff
@@ -568,6 +597,14 @@ async def async_get_mqtt(hass: HomeAssistant) -> dict[str, list[str]]:
     return mqtt
 
 
+@callback
+def async_register_preload_platform(hass: HomeAssistant, platform_name: str) -> None:
+    """Register a platform to be preloaded."""
+    preload_platforms: list[str] = hass.data[DATA_PRELOAD_PLATFORMS]
+    if platform_name not in preload_platforms:
```

Comment:
Should we do `assert platform_name not in preload_platforms` as it would be a bug if it's in there twice?

Reply:
When the base integration platforms get loaded (ie `group`) it would try to add them again so I added the guard